### PR TITLE
8258855: Two tests sun/security/krb5/auto/ReplayCacheTestProc.java and ReplayCacheTestProcWithMD5.java failed on OL8.3

### PR DESCRIPTION
**Pull request by fguallini**
Kerberos new replay cache format released in 1.18 (installed in OL8.3) is causing the tests failures:
`https://web.mit.edu/kerberos/www/krb5-latest/doc/formats/rcache_file_format.html`

New and old format are not compatible, that means they cannot share a rcache file. Since there is no interoperability between java GSS-API and native GSS-API new rcache format at the moment, this patch only enables the test scenarios that mimic AP-REQ replays with multiple processes using native GSS-API only (when available) and multiple processes using Java own implementation API only. 

If there is a decision to support the new format in the future, tests will be revisited accordingly.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8258855](https://bugs.openjdk.java.net/browse/JDK-8258855): Two tests sun/security/krb5/auto/ReplayCacheTestProc.java and ReplayCacheTestProcWithMD5.java failed on OL8.3

### Reviewers
 * [Weijun Wang](https://openjdk.java.net/census#weijun) (@wangweij - **Reviewer**) ⚠️ Review applies to abfa28babebc988f734c9528dcbe2ccbcdea3c18
 * [Rajan Halade](https://openjdk.java.net/census#rhalade) (@rhalade - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2676/head:pull/2676`
`$ git checkout pull/2676`

### CODE REVIEW

File: test/jdk/sun/security/krb5/auto/ReplayCacheTestProcWithMD5.java [hunk 1]
```diff
@@ -32,5 +32,11 @@
  * @run main jdk.test.lib.FileInstaller TestHosts TestHosts
  * @run main/othervm/timeout=300 -Djdk.krb5.rcache.useMD5=true
  *           -Djdk.net.hosts.file=TestHosts
- *           -Dtest.service=host ReplayCacheTestProc
+ *           -Dtest.service=host
+ *           -Dtest.libs=J ReplayCacheTestProc
+ * @run main/othervm/timeout=300 -Djdk.krb5.rcache.useMD5=true
+ *           -Djdk.net.hosts.file=TestHosts
+ *           -Dtest.service=host
+ *           -Dtest.libs=N ReplayCacheTestProc
```
Comment (wangweij) [inline test/jdk/sun/security/krb5/auto/ReplayCacheTestProcWithMD5.java:40]: No need to test native libs with MD5.

**Review by wangweij — APPROVED** [review]
Only one nit.

File: test/jdk/sun/security/krb5/auto/ReplayCacheTestProcWithMD5.java (hunk 1 above)
Comment (fguallini) [reply test/jdk/sun/security/krb5/auto/ReplayCacheTestProcWithMD5.java]: Done

File: test/jdk/sun/security/krb5/auto/ReplayCacheTestProc.java [hunk 2]
```diff
@@ -117,6 +119,15 @@ public static void main0(String[] args) throws Exception {
             Ex[] result;
             int numPerType = 2; // number of acceptors per type
 
+            // User-provided libs
+            String userLibs = System.getProperty("test.libs");
+            Asserts.assertNotNull(userLibs, "test.libs property must be provided");
+            libs = userLibs.split(",");
+            if (Arrays.asList(libs).contains("N") && !isNativeLibAvailable()) {
+                System.out.println("Native mode not available - skipped");
```
Comment (rhalade) [inline test/jdk/sun/security/krb5/auto/ReplayCacheTestProc.java:127]: Can you please add a comment here with which scenarios are skipped?

Comment (fguallini) [reply test/jdk/sun/security/krb5/auto/ReplayCacheTestProc.java:128]: Added an extra comment there.

**Review by rhalade — APPROVED** [review]

**Pull request closed without merging**